src/common/k8s-api/kube-json-api.ts

```diff
@@ -70,6 +73,20 @@ export interface KubeJsonApiError extends JsonApiError {
 }
 
 export class KubeJsonApi extends JsonApi<KubeJsonApiData> {
+  static forCluster(cluster: Cluster): KubeJsonApi {
+    const port = LensProxy.getInstance().port;
+
+    return new this({
+      serverAddress: `http://127.0.0.1:${port}`,
+      apiBase: apiKubePrefix,
+      debug: isDebugging,
+    }, {
+      headers: {
+        "Host": `${cluster.id}.localhost:${port}`,
```

Comment:
Would have used `x-cluster-id` instead but it didn't work, it was bypassing the `getProxyTarget` check in `LensProxy` 😕 that is why I removed the `x-cluster-id` handling from below as it seems to be legacy anyway.